synapse/cores/common.py

```diff
@@ -2873,8 +2873,8 @@ def delTufo(self, tufo):
             self.delTufoDset(tufo, name)
 
         if self.caching:
-            for prop, valu in list(tufo[1].items()):
-                self._bumpTufoCache(tufo, prop, valu, None)
+            for prop, _valu in list(tufo[1].items()):
```

Comment:
i'd typically use something like pval here so it's a bit more visually distinct, but really good catch.  I'm actually kinda suprised we haven't encountered more bugs with the caching...  Write-through caching always seems to be difficult to get right :)  Tweak that _valu name, and smoke it in.

Reply:
fixed. smashing.